### ISSUE TITLE
Select platform does not seem to work correctly

### ISSUE THREAD

**Issue by ershov-a** [rospogrigio/localtuya#689]
## The problem

`Select` platform does not seem to work correctly.
Got an error:
```
Logger: homeassistant.components.select
Source: custom_components/localtuya/select.py:42
Integration: Select(documentation, issues)
First occurred: 21:32:20 (1 occurrences)
Last logged: 21:32:20
Error while setting up localtuya platform for select

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 249, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/localtuya/common.py", line 74, in async_setup_entry
    entity_class(
  File "/config/custom_components/localtuya/select.py", line 42, in __init__
    self._valid_options = self._config.get(CONF_OPTIONS).split(";")
AttributeError: 'NoneType' object has no attribute 'split'
```

My device is Royal Clima Brezza RCB-150.

## Environment

- Localtuya version: 3.3.0  
- Last working localtuya version (if known and relevant): unknown, `select` platform implemented only in version 3.3.0 
- Home Assistant Core version: core-2021.12.1 
- [] Are you using the Home Assistant Tuya Cloud component ? 
- [] Are you using the Tuya App in parallel ? 

## Steps to reproduce

1. Obtain Tuya device's Key and ID.
2. Identify DPs.
3. Add Tuya device to Home Assistant config with following DPs:
- switch - Brezza state
- sensor - Brezza supply air
- switch - Brezza heater
- sensor - Brezza set temp
- switch - Brezza ionization
- sensor - Brezza mode
- sensor - Brezza fan
- select - Brezza mode select
4. Get an error in Home Assistant logs about `select` platfrom.

## Configuration `configuration.yaml` or `config_flow`

```yaml
localtuya:
  - host: ip
    device_id: id
    local_key: key
    friendly_name: Brezza
    protocol_version: "3.3"
    entities:
      - platform: switch
        friendly_name: Brezza state
        id: 1
      - platform: sensor
        friendly_name: Brezza supply air
        id: 22
        unit_of_measurement: "°C"
      - platform: switch
        friendly_name: Brezza heater
        id: 42
      - platform: sensor
        friendly_name: Brezza set temp
        id: 102
        unit_of_measurement: "°C"
      - platform: switch
        friendly_name: Brezza ionization
        id: 5
      - platform: sensor
        friendly_name: Brezza mode
        id: 2
      - platform: sensor
        friendly_name: Brezza fan
        id: 12
      - platform: select
        friendly_name: Brezza mode select
        id: 2
        select_options: Air_Supply;Recirculation;Sleep;Auto;Eco;Anti_Condensation
        select_options_friendly: Air_Supply;Recirculation;Sleep;Auto;Eco;Anti_Condensation
```

## DP dump

```
    DPS [1] VALUE [False]  - ON/OFF
    DPS [2] VALUE [Air_Supply] - mode: Air_Supply, Recirculation, Sleep, Auto, Eco, Anti_Condensation
    DPS [3] VALUE [0] - PM2.5 value
    DPS [5] VALUE [False] - ionization
    DPS [11] VALUE [96] - filter resource (0-100)
    DPS [12] VALUE [1] - supply fan speed (1-6)
    DPS [16] VALUE [cancel] - ?
    DPS [22] VALUE [-3] - supply air temperature 
    DPS [42] VALUE [False] - heater
    DPS [101] VALUE [cancel] - ?
    DPS [102] VALUE [22] - set temperature
```

## Provide Home Assistant taceback/logs

```txt
2022-01-02 21:32:17 WARNING (SyncWorker_0) [homeassistant.loader] We found a custom integration localtuya which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2022-01-02 21:32:19 DEBUG (MainThread) [custom_components.localtuya.discovery] Listening to broadcasts on UDP port 6666 and 6667
2022-01-02 21:32:19 DEBUG (MainThread) [custom_components.localtuya.sensor] [id] Adding sensor.brezza_supply_air with configuration: {'platform': 'sensor', 'friendly_name': 'Brezza supply air', 'id': 22, 'unit_of_measurement': '°C'}
2022-01-02 21:32:19 DEBUG (MainThread) [custom_components.localtuya.sensor] [id] Adding sensor.brezza_set_temp with configuration: {'platform': 'sensor', 'friendly_name': 'Brezza set temp', 'id': 102, 'unit_of_measurement': '°C'}
2022-01-02 21:32:19 DEBUG (MainThread) [custom_components.localtuya.sensor] [id] Adding sensor.brezza_mode with configuration: {'platform': 'sensor', 'friendly_name': 'Brezza mode', 'id': 2}
2022-01-02 21:32:19 DEBUG (MainThread) [custom_components.localtuya.sensor] [id] Adding sensor.brezza_fan with configuration: {'platform': 'sensor', 'friendly_name': 'Brezza fan', 'id': 12}
2022-01-02 21:32:19 DEBUG (MainThread) [custom_components.localtuya.switch] [id] Adding switch.brezza_state with configuration: {'platform': 'switch', 'friendly_name': 'Brezza state', 'id': 1}
2022-01-02 21:32:19 DEBUG (MainThread) [custom_components.localtuya.switch] [id] Adding switch.brezza_heater with configuration: {'platform': 'switch', 'friendly_name': 'Brezza heater', 'id': 42}
2022-01-02 21:32:19 DEBUG (MainThread) [custom_components.localtuya.switch] [id] Adding switch.brezza_ionization with configuration: {'platform': 'switch', 'friendly_name': 'Brezza ionization', 'id': 5}
2022-01-02 21:32:20 ERROR (MainThread) [homeassistant.components.select] Error while setting up localtuya platform for select
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 249, in _async_setup_platform
await asyncio.shield(task)
File "/config/custom_components/localtuya/common.py", line 74, in async_setup_entry
entity_class(
File "/config/custom_components/localtuya/select.py", line 42, in __init__
self._valid_options = self._config.get(CONF_OPTIONS).split(";")
AttributeError: 'NoneType' object has no attribute 'split'
```

**Comment by Zen3515** [comment]
I don't know if I should raise a new issue, but in my case, the value returned from DP is int, so, when it tries to map state to select option, it doesn't work, list of str, but the value is int.

```yaml
      - platform: select
        friendly_name: RCBO Power On State
        id: 38
        select_options: 0;1;2
        select_options_friendly: Off;On;Last State
```

I have to fix it by casting state to str `state = str(self.dps(self._dp_id))`
```python

    def status_updated(self):
        """Device status was updated."""
        state = str(self.dps(self._dp_id))
        print(
            f"self._dp_id={self._dp_id}, state={state}, self._valid_options={self._valid_options}"
        )
        self._state_friendly = self._display_options[self._valid_options.index(state)]
        self._state = state
```

this is what it print
self._dp_id=38, state=2, self._valid_options=['0', '1', '2']
